presto-main/src/test/java/com/facebook/presto/operator/TestAnnotationEngine.java

```diff
@@ -0,0 +1,884 @@
+import static com.facebook.presto.operator.AnnotationBasedOperators.ExactAggregationFunction;
+import static com.facebook.presto.operator.AnnotationBasedOperators.ExplicitSpecializedAggregationFunction;
+import static com.facebook.presto.operator.AnnotationBasedOperators.MultiScalarFunction;
+import static com.facebook.presto.operator.AnnotationBasedOperators.ParametricScalarFunction;
+import static com.facebook.presto.operator.AnnotationBasedOperators.SimpleInjectionScalarFunction;
+import static com.facebook.presto.operator.AnnotationBasedOperators.StaticMethodScalarFunction;
+import static com.facebook.presto.operator.AnnotationBasedOperators.WithNullableComplexArgScalarFunction;
+import static com.facebook.presto.operator.AnnotationBasedOperators.WithNullablePrimitiveArgScalarFunction;
+import static com.facebook.presto.operator.aggregation.AggregationFromAnnotationsParser.parseFunctionDefinition;
+import static com.facebook.presto.operator.aggregation.AggregationFromAnnotationsParser.parseFunctionDefinitions;
+import static com.facebook.presto.spi.type.StandardTypes.ARRAY;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestAnnotationEngine
+{
+    @Test
+    public void testSimpleExactAggregationParse()
+            throws Exception
+    {
+        Signature expectedSignature = new Signature(
+                "simple_exact_aggregate",
+                FunctionKind.AGGREGATE,
+                DoubleType.DOUBLE.getTypeSignature(),
+                ImmutableList.of(DoubleType.DOUBLE.getTypeSignature()));
+
+        ParametricAggregation aggregation = parseFunctionDefinition(ExactAggregationFunction.class);
+        assertEquals(aggregation.getDescription(), "Simple exact aggregate description");
+        assertTrue(aggregation.isDeterministic());
+        assertEquals(aggregation.getSignature(), expectedSignature);
+        ParametricImplementationsGroup<AggregationImplementation> implementations = aggregation.getImplementations();
+        assertEquals(implementations.getExactImplementations().size(), 1);
+        assertEquals(implementations.getGenericImplementations().size(), 0);
+        assertEquals(implementations.getSpecializedImplementations().size(), 0);
+        AggregationImplementation implementation = (AggregationImplementation) implementations.getExactImplementations().values().toArray()[0];
+        assertFalse(implementation.getStateSerializerFactory().isPresent());
+        assertEquals(implementation.getDefinitionClass(), ExactAggregationFunction.class);
+        assertEquals(implementation.getInputDependencies().size(), 0);
+        assertEquals(implementation.getCombineDependencies().size(), 0);
+        assertEquals(implementation.getOutputDependencies().size(), 0);
```

Comment:
Add a helper method that does

```
assertDependencyCount(implementation, int input, int combine, int output)
{
    assertEquals(implementation.getInputDependencies().size(), input);
    assertEquals(implementation.getCombineDependencies().size(), combine);
    assertEquals(implementation.getOutputDependencies().size(), output);
}
```

If you see other opportunities to reduce boilerplate, consider taking it.